lua/metals/config.lua

```diff
@@ -37,24 +37,45 @@ local function in_disabled_mode(config)
   return disabled
 end
 
+--- Used to generate Metals installation file name depending on configuration
+--- @return string fileame
+local function metals_filename()
+  if config_cache and config_cache.settings.metals.useGlobalExecutable then
+    return metals_name
+  else
+    return Path:new(util.nvim_metals_cache_dir, metals_name).filename
+  end
+end
+
 --- Ultimately what will be passed to the config.cmd to initialize the LSP
 --- connection. If a user is using useGlobalExecutable then we just default
 --- to `metals`, if not we take control and construct the location in the
 --- cache dir.
 --- @return string executable
 local function metals_bin()
+  local metals_filename = metals_name
+  -- Coursier will install both a "metals" and "metals.bat", on Windows we should use "metals.bat" as exe
```

Comment:
Out of curiosity, why does it install a `metals` on windows then if it needs to use `metals.bat`?

Reply:
I think is a linux exe, or equivalent, has some bash in the header and the rest is binary